WooCommerce/Classes/POS/Presentation/Reusable Views/POSSendReceiptView.swift

```diff
@@ -30,13 +36,27 @@ struct POSSendReceiptView: View {
                 .font(POSFontStyle.posTitleRegular)
                 .focused()
                 .padding()
+            if let errorMessage = errorMessage {
+                Text(errorMessage)
+                    .font(POSFontStyle.posBodyRegular)
+                    .foregroundColor(.red)
+            }
 
             Button(action: {
                 Task { @MainActor in
+                    guard isEmailValid else {
+                        errorMessage = Localization.emailValidationErrorText
+                        return
+                    }
                     isLoading = true
-                    await posModel.sendReceipt(to: textFieldInput)
+                    do {
+                        try await posModel.sendReceipt(to: textFieldInput)
+                        isShowingSendReceiptView = false
+                        errorMessage = nil
```

Comment:
We could consider setting it to `nil` before sending the receipt as well. To have this clean view when we're resubmitting after the failure.

Reply:
Great detail, much better to clear it without having to wait for the callback. Changed on https://github.com/woocommerce/woocommerce-ios/pull/14651/commits/869e6049ae6f9ffd3ba3baaab30f1c421e35feed